java-connectors/kafka-connect-azure-eventhubs/src/test/java/io/lenses/streamreactor/connect/azure/eventhubs/source/EventHubsKafkaConsumerControllerTest.java

```diff
@@ -0,0 +1,175 @@
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.ArrayBlockingQueue;
+import org.apache.kafka.clients.consumer.ConsumerRecord;
+import org.apache.kafka.clients.consumer.ConsumerRecords;
+import org.apache.kafka.common.header.Header;
+import org.apache.kafka.common.header.Headers;
+import org.apache.kafka.connect.data.Schema;
+import org.apache.kafka.connect.source.SourceRecord;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+class EventHubsKafkaConsumerControllerTest {
+
+  private static final String INPUT_TOPIC = "INPUT";
+  private static final String OUTPUT_TOPIC = "OUTPUT";
+  private static final String INPUT_TOPIC_2 = "INPUT2";
+  private static final String OUTPUT_TOPIC_2 = "OUTPUT2";
+  public static final int DEFAULT_CAPACITY = 10;
+  private ArrayBlockingQueue<ConsumerRecords<byte[], byte[]>> recordsQueue;
+
+  private EventHubsKafkaConsumerController testObj;
+
+
+  @BeforeEach
+  void setUp() {
+    recordsQueue = new ArrayBlockingQueue<>(DEFAULT_CAPACITY);
+  }
+
+  @Test
+  void pollShouldPollQueueAndReturnSourceRecords() throws InterruptedException {
+    //given
+    Duration duration = Duration.of(2, ChronoUnit.SECONDS);
+    Map<String, String> inputOutputMap = Map.of(INPUT_TOPIC, OUTPUT_TOPIC);
+
+    SourceDataType mockedDataType = mock(SourceDataType.class);
+    when(mockedDataType.getSchema()).thenReturn(Schema.OPTIONAL_STRING_SCHEMA);
+
+    KeyValueTypes mockedKeyValueTypes = mock(KeyValueTypes.class);
+    when(mockedKeyValueTypes.getKeyType()).thenReturn(mockedDataType);
+    when(mockedKeyValueTypes.getValueType()).thenReturn(mockedDataType);
+
+    KafkaByteBlockingQueuedProducer mockedBlockingProducer = mock(
+        KafkaByteBlockingQueuedProducer.class);
+    when(mockedBlockingProducer.getKeyValueTypes()).thenReturn(mockedKeyValueTypes);
+
+    Headers headersMock = mock(Headers.class);
+    List<Header> emptyHeaderList = Collections.emptyList();
+    when(headersMock.iterator()).thenReturn(emptyHeaderList.iterator());
+
+    ConsumerRecord consumerRecord = mock(ConsumerRecord.class);
+    when(consumerRecord.topic()).thenReturn(INPUT_TOPIC);
+    when(consumerRecord.headers()).thenReturn(headersMock);
+    List<ConsumerRecord<String, String>> consumerRecordList = Collections.singletonList(consumerRecord);
+
+    ConsumerRecords mockedRecords = mock(ConsumerRecords.class);
+    when(mockedRecords.count()).thenReturn(consumerRecordList.size());
+    when(mockedRecords.iterator()).thenReturn(consumerRecordList.iterator());
+    recordsQueue.put(mockedRecords);
+
+    //when
+    testObj = new EventHubsKafkaConsumerController(mockedBlockingProducer, recordsQueue,
+        inputOutputMap);
+    List<SourceRecord> sourceRecords = testObj.poll(duration);
+
+    //then
+    verify(mockedBlockingProducer).start();
+    verify(mockedDataType, times(2)).getSchema();
+    verify(mockedBlockingProducer, times(2)).getKeyValueTypes();
```

Comment:
Why 2? No context on this number.  Please add an assertion hint.

Reply:
There's no assertion so I cannot hint that. I changed this test a bit so it should be more straightforward around that part, hope it'll help!